OrbitCore/Capture.cpp

```diff
@@ -100,6 +100,7 @@ void Capture::ClearCaptureData() {
 void Capture::PreFunctionHooks() {
   GSelectedInCaptureFunctions = GetSelectedFunctions();
 
+  GSelectedFunctionsMap.clear();
```

Comment:
TBH - I do not understand what sequence of events leads to the situation where selected functions is nullptr, and why calling clear here fixes it. can you please explain it in the commit message?

Reply:
Steps to reproduce:

1. Start Orbit -> hook function(s) -> start capture -> save capture
2. Restart Orbit ans select process (but don't hook functions)
3. Load the capture from the first step (we fill `Capture::GSelectedFunctionsMap` on this step)
4. Start a new capture. Capture::GSelectedInCaptureFunctions is empty (we updated it 2 lines above) when GSelectedFunctionsMap is not empty (we didn't clear it in any point and it's still have data from the saved capture)

So when we start a capture GSelectedInCaptureFunctions and GSelectedFunctionsMap should be synchronized, but in case described above we see discrepancy. We won't have this situation if open capture without connection.

I updated commit message too.

Reply:
Maybe a bit more detail to add here:
`GetSelectedFunctions()` basically runs over all loaded functions of a process and check whether it is currently in the `GSelectedFunctionsMap` or a special Orbit function. If one of this is the case, it will be returned.
So if your process does not have any functions (because you loaded the capture and did nothing else before), selected functions will be empty, but the map still contains the functions from loading the capture.